cubedash/_stac.py

```diff
@@ -266,7 +266,40 @@ def as_stac_item(dataset: DatasetItem) -> pystac.Item:
     return item
 
 
+def _provider_from_dict(d: dict) -> pystac.Provider:
+    """A custom parser that extends pystac.Provider.from_dict()"""
+    # pystac.Provider.from_dict() does not validate roles against ProviderRole nor create a list
+    p = pystac.Provider.from_dict(d)
+    roles = p.roles  # unvalidated
+    if roles is not None:
+        p.roles = []  # reset
+        if isinstance(roles, str):
+            roles = [roles]
+        for r in roles:
+            if r is None:
+                continue
+            try:
+                p.roles.append(pystac.ProviderRole(r))
+            except ValueError as e:
+                _LOG.warning(f"Could not parse Provider role: {e}")
+        if len(p.roles) == 0:
+            p.roles = None
+    return p
+
+
 def as_stac_collection(res: CollectionItem) -> pystac.Collection:
+    """Return a pystac.Collection view of an internal CollectionItem"""
+    # Add Providers, Summaries and Keywords
+    # providers: list[dict[str, Any]] | None -> list[pystac.Provider]
+    # summaries: dict[str, Any] | None -> pystac.Summaries
+    # keywords: list[str] | None
+    providers = None
```

Comment:
If you change this into:
```python
providers = None if res.providers is None else [_provider_from_dict(xx) for ..]
```
you can avoid the negated if-condition, and the "final" value gets assigned to providers right away.

Same comment for summaries a couple of lines below.